### PR TITLE
[sqlite3] Update to 3.42.0

### PR DESCRIPTION
**Pull request by Thomas1664**
If this PR updates an existing port, please uncomment and fill out this checklist:

- [x] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md)
- [x] SHA512s are updated for each updated download
- [x] The "supports" clause reflects platforms that may be fixed by this new version
- [x] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [x] Only one version is added to each modified port's versions file.

### CODE REVIEW

**Comment by dg0yt** [comment]
Do you feel able to write a vcpkg-cmake-wrapper for the multi-config stuff, similar to port zlib?
It might even leverage the unofficial CMake config.
It would make patching as in sqlitepp obsolete.

**Comment by Thomas1664** [comment]
> Do you feel able to write a vcpkg-cmake-wrapper for the multi-config stuff, similar to port zlib? It might even leverage the unofficial CMake config. It would make patching as in sqlitepp obsolete.
> 
> Cf. #31659

I tried but I'd like to point out that this is my first wrapper.

**Comment by Thomas1664** [comment]
@dg0yt I don't know why the wrapper only works on Windows. CMake should find libsqlite3.a as well since it adds the lib prefix on UNIX.

**Comment by huangqinjin** [comment]
@dg0yt Shall we remove json1 feature?  It is builtin since 3.38 https://sqlite.org/releaselog/3_38_0.html

**Comment by dg0yt** [comment]
> @dg0yt Shall we remove json1 feature? It is builtin since 3.38 https://sqlite.org/releaselog/3_38_0.html

Not my decision. I would always prefer a grace period with a warning, but other PRs didn't care for that.

**Comment by MonicaLiu0311** [comment]
> > @dg0yt Shall we remove json1 feature? It is builtin since 3.38 https://sqlite.org/releaselog/3_38_0.html
> 
> Not my decision. I would always prefer a grace period with a warning, but other PRs didn't care for that.

Please remove unused features, thanks.

File: ports/sqlite3/portfile.cmake [hunk 1]
```diff
@@ -25,24 +25,6 @@ else()
     set(SQLITE_API "")
 endif()
 
-vcpkg_check_features(OUT_FEATURE_OPTIONS Unused
```
Comment (Thomas1664) [inline ports/sqlite3/portfile.cmake:28]: @MonicaLiu0311 I also removed all of the other unused features

Comment (dg0yt) [reply ports/sqlite3/portfile.cmake:28]: You must revert this removal and the following changes. Only this *variable* is unused. But the function has side effects: The listed options are also set as variables and take effect via `configure_file`. Use git blame to find the related PRs.

**Comment by dg0yt** [comment]
> Shall we remove json1 feature? It is builtin since 3.38 https://sqlite.org/releaselog/3_38_0.html

This seems to need a default feature with inverted `SQLITE_OMIT_JSON` now.

**Comment by Thomas1664** [comment]
> > Shall we remove json1 feature? It is builtin since 3.38 https://sqlite.org/releaselog/3_38_0.html
> 
> This seems to need a default feature with inverted `SQLITE_OMIT_JSON` now.

You have to decide what you want. I was asked to remove the feature and now I should add it back in?

**Comment by MonicaLiu0311** [comment]
[vcpkg.json#json1](https://github.com/microsoft/vcpkg/pull/31703/files#diff-e826c635d164da06c36f48dc9469b8d5130bf288c8b5ce1dcf82dd071af3d94cR33-L36)

Also needs to be removed in `vcpkg.json`.

**Comment by dg0yt** [comment]
> > > Shall we remove json1 feature? It is builtin since 3.38 https://sqlite.org/releaselog/3_38_0.html
> > 
> > 
> > This seems to need a default feature with inverted `SQLITE_OMIT_JSON` now.
> 
> You have to decide what you want. I was asked to remove the feature and now I should add it back in?

@Thomas1664 You originally explained that it "is builtin since 3.38", and I took this for non-optional. Now that you broke the other features, I read the linked release notes and learned that it is still optional, just enabled by default.
Based on this new information, I made a deliberate recommendation to correctly model this optionality.

**Comment by Thomas1664** [comment]
Should the feature become a default feature? Upstream enables it by default.

**Comment by dg0yt** [comment]
> Should the feature become a default feature? Upstream enables it by default.

The "default" was part of my recommendation, yes.

**Comment by Thomas1664** [comment]
Note: build failures not caused by this PR. Failures are due to authentication issues with binary cache.

**Comment by MonicaLiu0311** [comment]
All features are tested successfully in the following triplet:
```
x86-windows
x64-windows
x64-windows-static
```
The usage test passed (header files found):
```
sqlite3 provides CMake targets:

    find_package(unofficial-sqlite3 CONFIG REQUIRED)
    target_link_libraries(main PRIVATE unofficial::sqlite3::sqlite3)
```

**Review by MonicaLiu0311 — APPROVED** [review]

**Pull request merged**